src/SmoFacade/Server.cs

```diff
@@ -130,24 +142,24 @@ private DefaultFileLocations DefaultFileLocations()
     /// </summary>
     /// <param name="backupDirectoryQuery"></param>
     /// <returns></returns>
-    private string BackupDirectoryOrDefault(string backupDirectoryQuery)
-    {
-      string result = null;
-      if(!string.IsNullOrWhiteSpace(backupDirectoryQuery)) {
-        using var cmd = SqlConnection.CreateCommand();
-        cmd.CommandText = backupDirectoryQuery;
-        using var reader = cmd.ExecuteReader();
-        if(reader.Read())
-          result = (string)reader[0];
-      }
-
-      result = result ?? _server.BackupDirectory;
-      if(result.EndsWith("\\") || result.EndsWith("/"))
-        result = result.Substring(0, result.Length - 1);
-
-      return result;
-    }
-
+    private string BackupDirectoryOrDefault(string backupDirectoryQuery)
+    {
+      string result = null;
+      if(!string.IsNullOrWhiteSpace(backupDirectoryQuery)) {
+        using var cmd = SqlConnection.CreateCommand();
+        cmd.CommandText = backupDirectoryQuery;
+        using var reader = cmd.ExecuteReader();
+        if(reader.Read())
+          result = (string)reader[0];
+      }
+
+      result ??= _server.BackupDirectory;
```

Comment:
Oooo, I didn't know you could do this - `??=`

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator